py/selenium/webdriver/support/expected_conditions.py

```diff
@@ -43,6 +43,7 @@
 T = TypeVar("T")
 
 WebDriverOrWebElement = Union[WebDriver, WebElement]
+attr_get_property = ["value", "checked", "index", "selected"]
```

Comment:
The "id" should be a property too, otherwise you only read ids defined in html.
All the `.get_dom_attribute("id")` calls below should be replaced by `get_property("id")`